airflow/dag_processing/bundles/base.py

```diff
@@ -112,3 +125,22 @@ def view_url(self, version: str | None = None) -> str | None:
         :param version: Version to view
         :return: URL to view the bundle
         """
+
+    @contextmanager
+    def lock(self):
+        if self._locked:
```

Comment:
is there a need to tolerate nested locking? should we maybe raise instead?

Reply:
For now, yes. And probably should, yes. Gits initialize calls refresh though. And I'm trying to keep it simple. Happy to refactor in a followup.